homeassistant/components/owlet/__init__.py

```diff
@@ -0,0 +1,83 @@
+"""
+Support for Owlet baby monitors.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/owlet/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+
+REQUIREMENTS = ['pyowlet==1.0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'owlet'
+
+SENSOR_TYPES = [
+    'oxygen_level',
+    'heart_rate',
+    'base_station_on',
+    'movement'
+]
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up owlet component."""
+    from pyowlet.PyOwlet import PyOwlet
+
+    username = config[DOMAIN].get(CONF_USERNAME)
+    password = config[DOMAIN].get(CONF_PASSWORD)
+    name = config[DOMAIN].get(CONF_NAME)
+
+    device = PyOwlet(username, password)
```

Comment:
We should log error and return False from setup if credentials are bad. Please validate that via the library.